tests/components/pooldose/test_sensor.py

```diff
@@ -156,39 +144,128 @@ async def test_sensor_entity_unavailable_no_coordinator_data(
 
 
 @pytest.mark.usefixtures("mock_pooldose_client")
-async def test_temperature_sensor_dynamic_unit(
+async def test_temperature_sensor_unit_from_api(
     hass: HomeAssistant,
     mock_config_entry: MockConfigEntry,
     mock_pooldose_client: AsyncMock,
-    freezer: FrozenDateTimeFactory,
 ) -> None:
-    """Test temperature sensor uses dynamic unit from API data."""
+    """Test temperature sensor correctly reads unit from API data."""
     mock_config_entry.add_to_hass(hass)
-
     await hass.config_entries.async_setup(mock_config_entry.entry_id)
     await hass.async_block_till_done()
 
-    # Verify initial Celsius unit
+    # Verify sensor correctly uses unit from API data
     temp_state = hass.states.get("sensor.pool_device_temperature")
-    assert temp_state.attributes[ATTR_UNIT_OF_MEASUREMENT] == UnitOfTemperature.CELSIUS
+    assert temp_state is not None
+    assert temp_state.state == "25"
+    assert temp_state.attributes["unit_of_measurement"] == "°C"
+
+
+@pytest.mark.usefixtures("mock_pooldose_client")
+async def test_flow_rate_sensor_unit_from_api(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_pooldose_client: AsyncMock,
+) -> None:
+    """Test flow rate sensor gets unit from API data."""
+    mock_config_entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    # Verify sensor uses unit from fixture data
+    flow_state = hass.states.get("sensor.pool_device_flow_rate")
+    assert flow_state is not None
+    assert flow_state.attributes["unit_of_measurement"] == "L/s"
+    assert flow_state.state == "150"
```

Comment:
In theory these are tested by the snapshot test right?

Reply:
fixed in https://github.com/home-assistant/core/pull/156002/commits/c56811cd172f101ae7fe686414aeef9dd698befb